### PR TITLE
Add dc-controlplane/services modules and extend bootstrap

### PR DESCRIPTION
**Pull request by HiranAdikari**
## Summary

- **dc-controlplane module**: Wraps the RKE2 cluster, Harvester NAD, and LB IPPool for the DC-API control plane. Includes a `kubectl` patch that sets `selector.scope` on the IPPool (workaround until the Harvester provider exposes the field declaratively). Accepts `harvester_kubeconfig_path` to run the patch after cluster creation.
- **dc-controlplane-services module**: Wraps all workloads running on the dcapi cluster — PostgreSQL StatefulSet, DC-API Deployment + Ingress, ARC controller + runner scale set, and all Kubernetes Secrets. Environment layers become thin callers with no inline resource definitions.
- **bootstrap module**: Add VM image and storage management — new variables for image source URL, disk size, and storage class; cloud-init template extended with additional packages and disk setup; outputs expose the provisioned image ID for downstream layers.
- **namespace-credential-provisioner**: Improve the reconcile script to handle edge cases and surface clearer error messages on failure.

## Test plan

- [ ] `terraform validate` passes in `02-dc-controlplane` and `02-dc-controlplane-services` layers (lk-dev)
- [ ] `terraform plan` produces no unexpected diff after modularisation
- [ ] `terraform apply` on lk-dev brings up dcapi cluster + all services cleanly from scratch

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: modules/management/dc-controlplane-services/main.tf [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+              port {
+                number = 80
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (HiranAdikari) [reply modules/management/dc-controlplane-services/main.tf:453]: Acknowledged. TLS termination for the DC-API ingress is intentionally deferred — this endpoint is internal-only () and not exposed beyond the management network. TLS will be added as a dedicated follow-up once cert-manager is wired into the dcapi cluster.

File: modules/management/harvester-integration/main.tf [hunk 2]
```diff
@@ -110,6 +110,54 @@ resource "kubernetes_secret" "rancher_credential_token" {
...
+        }
+        prometheus 0.0.0.0:9153
+        forward . /etc/resolv.conf
+        cache 30
+        loop
+        reload
+        loadbalance
+    }
+  EOT
+}
```
[the remark this answers is not in the document]
Comment (HiranAdikari) [reply modules/management/harvester-integration/main.tf:159]: Intentional design, not a bug. The previous `count = 0` approach would destroy the entire `Corefile` key on the ConfigMap when `patch_coredns = false`, which breaks CoreDNS entirely. The always-present resource writes one of two templates: the patched Corefile (with the Rancher LB hosts entry) when `patch_coredns = true`, or the stock default Corefile when `patch_coredns = false`. The default template matches the rke2-coredns chart output exactly — verified against the live cluster before merging.

File: modules/management/dc-controlplane-services/main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (HiranAdikari) [reply modules/management/dc-controlplane-services/main.tf:453]: Go ahead

**Review by gnudeep — APPROVED** [review]
